pkg/test/controller/reconciler/testreconciler.go

```diff
@@ -142,12 +142,12 @@ func (r *TestReconciler) ReconcileObjectMeta(ctx context.Context, om metav1.Obje
 }
 
 // Creates and reconciles all unstructureds in the unstruct list. Returns a cleanup function that should be defered immediately after calling this function.
-func (r *TestReconciler) CreateAndReconcile(ctx context.Context, unstructs []*unstructured.Unstructured, cleanupPolicy ResourceCleanupPolicy) func() {
+func (r *TestReconciler) CreateAndReconcile(ctx context.Context, dependencies []*unstructured.Unstructured, cleanupPolicy ResourceCleanupPolicy, mainResource *unstructured.Unstructured) func() {
 	r.t.Helper()
-	cleanupFuncs := make([]func(), 0, len(unstructs))
-	for _, u := range unstructs {
+	cleanupFuncs := make([]func(), 0, len(dependencies))
+	for _, u := range dependencies {
 		if err := r.mgr.GetClient().Create(ctx, u); err != nil {
-			r.t.Fatalf("error creating resource '%v': %v", u.GetKind(), err)
+			r.t.Fatalf("error creating dependecy '%v' for resource '%v/%v': %v", u.GetKind(), mainResource.GetName(), mainResource.GetKind(), err)
 		}
```

Comment:
No real objection, but don't we know from the test name?

Reply:
My experience with the post submit logs was that sometimes they are out of order. Another use case for this (at least for my debugging) is grep-ing against "error" or some other string and gleaning as much information as possible.